test/remote/recovery_key_tests.js

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const assert = require('insist')
+const config = require('../../config').getProperties()
+const crypto = require('crypto')
+const TestServer = require('../test_server')
+const Client = require('../client')()
+const hkdf = require('../../lib/crypto/hkdf')
+const Promise = require('bluebird')
+const jose = require('node-jose')
+
+describe('remote recovery keys', function () {
+  let server, client, email
+  const password = '(-.-)Zzz...'
+
+  let recoveryKeyId
+  let recoveryData
+  let keys
+  let keystore
+
+  function createRecoveryKey(uid, kB) {
```

Comment:
@shane-tomlinson This function does what the content-server *should* be doing when it comes to encryption. It might be a little helpful seeing how it pieces together. To be fair though, some of these parts could still be changed.